### PR TITLE
darwin.CoreFoundation: update to swift newest version

### PR DESCRIPTION
**Pull request by holymonson**
###### Motivation for this change
This PR is almost done, but it needs libdispatch update to macos-10.13 or above to provide `DISPATCH_APPLY_AUTO`. (#101229)
Also, it may need a newer ICU. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/darwin/swift-corelibs/foundation/default.nix [hunk 1]
```diff
@@ -10,71 +10,70 @@ let
...
-    rev    = "71aaba20e1450a82c516af1342fe23268e15de0a";
-    sha256 = "17kpql0f27xxz4jjw84vpas5f5sn4vdqwv10g151rc3rswbwln1z";
+    rev    = "${version}-RELEASE";
+    sha256 = "1fd0djza84rdh4jw7kw0kgm22r7jmga6zx1drzpymvpqyscxqsdq";
   };
 
-  nativeBuildInputs = [ ninja python3 ];
-  buildInputs = [ curl libxml2 objc4 ICU ];
+  patches = [
+    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/swift-corelibs/foundation/default.nix:25]: Can you add comments why we need this?

**Comment by supermarin** [comment]
@holymonson @SuperSandro2000 can we move this forward and update Swift to 5.5?

**Comment by holymonson** [comment]
> @holymonson @SuperSandro2000 can we move this forward and update Swift to 5.5?

Seems not, we are still waiting for a new macos SDK (#116341 #101229).

**Comment by supermarin** [comment]
Sounds good, will keep an  eye on the other ones

**Pull request closed without merging**